contracts/v-testnet-launch/GrtPool.sol

```diff
@@ -36,34 +36,31 @@ contract GrtPool is OwnableUpgradeable, GrtOffer, UUPSUpgradeable {
 
     function _authorizeUpgrade(address) internal override onlyOwner onlyProxy {}
 
-    receive() external payable {}
+    error TransferedAmountMustBePositive();
+    error ZeroAddressIsNotAllowed();
+    error OfferInactive();
+    error FailedToSendNativeTokens();
 
     function depositETHAndAcceptOffer(
         bytes32 offerId,
         address destAddr
     ) external payable returns (bytes32) {
-        require(
-            msg.value > 0,
-            "Grindery Pool: transfered amount must be positive."
-        );
-        require(
-            destAddr != address(0),
-            "Grindery Pool: zero address as destination address is not allowed."
-        );
-        require(
-            _offers[offerId].isActive,
-            "Grindery Pool: the offer is inactive."
-        );
-
+        if(msg.value <= 0)
+            revert TransferedAmountMustBePositive();
+        if(destAddr == address(0))
+            revert ZeroAddressIsNotAllowed();
+        if(!_offers[offerId].isActive)
+            revert OfferInactive();
         (bool sent, ) = address(this).call{value: msg.value}("");
```

Comment:
Maybe we can move this call to the bottom of the function to protect against reentrancy ?
And also to follow Solidity security Guidelines with Checks-Effects-Interactions pattern.
 https://docs.soliditylang.org/en/v0.6.11/security-considerations.html